### DISCUSSION TITLE
Add EDDTableFromParquet reader?

### DISCUSSION THREAD

**Discussion by MathewBiddle** [ERDDAP/erddap#194] [Ideas]
I'm continually seeing more and more groups using the Parquet Data File format (https://parquet.apache.org/docs/). Looking through the docs, I see there is an apache/parquet-java package (https://github.com/apache/parquet-java) which could lend itself to the java backend of ERDDAP. I'm curious if others have explored Parquet more in depth and if there is a need for ERDDAP to have the capacity to read from that format.

I haven't seen any discussion on the topic yet. 
https://github.com/search?q=org%3AERDDAP%20parquet&type=code
https://groups.google.com/g/erddap/search?q=parquet

**Comment by MathewBiddle** [comment]
Also, maybe a **writer** is part of the conversation too??

**Comment by srstsavage** [comment]
+1, very performant format for certain use cases, would be great to add support for it

**Comment by rmendels** [comment]
Parquet JDBC - https://www.cdata.com/drivers/parquet/jdbc/

**Comment by rmendels** [comment]
see also https://cdn.cdata.com/help/HIK/jdbc/

**Comment by kevin-obrien** [comment]
I'll add another +1 for this.

**Comment by rmendels** [comment]
Does anyone have sample Parquet files?

**Reply by MathewBiddle** [reply]
😨 https://obis-datasets.ams3.digitaloceanspaces.com/exports/obis_20240723.parquet

**Reply by MathewBiddle** [reply]
It's big tho! GBs

**Reply by srstsavage** [reply]
[glider.parquet.gz](https://github.com/user-attachments/files/16820698/glider.parquet.gz)

**Comment by srstsavage** [comment]
FWIW duckdb is very powerful/useful for exploring and analyzing parquet files, and also provides a JDBC interface:

https://duckdb.org/docs/api/java.html

ClickHouse is another great analysis platform that uses parquet. Lots of options!

**Comment by rmendels** [comment]
Thanks for the links.  Yes I already use duckdb for similar and knew it could do Parquet.  In R there is duckdbplyr which ties duckdb to the tidyverse.  I want to test if the JDBC already in ERDDAP can be made to work as is.  But that only gives you read,  not write,  a brief look at the Apache Parquet Java suggests it may not be so simple.  Also how much actual demand is there for this -  at least in our immediate communities.  Also,  though it is not my favorite organization, https://geoparquet.org.  This is related to GeoArrow,  see https://geoarrow.org  - scroll down for the discussion on the relationship with GeoParquet

**Comment by ChrisJohnNOAA** [comment]
I made an issue to track work on this. #196

**Reply by ChrisJohnNOAA** [reply]
I think directly using [parquet-java](https://github.com/apache/parquet-java) in a new dataset type is the best plan for this. My guess is it will likely be the most performant option.
